janus-idp/backstage-showcase
# feat: add `scaffoldedFrom` links between catalog entities and scaffolder template entities

## Description

- Added a custom processor to introduce the `scaffoldedFrom` relation between scaffolder templates and their generated entities.
- TODO: Add Custom Action to apply the `spec.scaffoldedFrom` field to the generated entities

![image](https://github.com/janus-idp/backstage-showcase/assets/50030060/ff469405-6bd6-4452-8468-eecc9a873d2c)

## Which issue(s) does this PR fix

- Fixes [RHIDP-2156](https://issues.redhat.com/browse/RHIDP-2156)


## PR acceptance criteria

Please make sure that the following steps are complete:


## How to test changes / Special notes to the reviewer
- Ingest an entity with a `spec.scaffoldedFrom` field that matches the `entityRef` of a template entity in the catalog (ex: `template:default/test-template`)
- Observe the relation graph of the entity in the catalog (or view the raw YAML/JSON of the entity and find the `scaffoldedFrom` relation in the `relations` array)
